### PR TITLE
[11.0][MIG] web_advanced_search_wildcard

### PR DESCRIPTION
**Pull request by JBF91**
Migration of web_advanced_search_wildcard.

No problems found, and no errors when using it, I thinks it's done.

### CODE REVIEW

**Comment by lfreeke** [comment]
:+1: (functional test)

File: web_advanced_search_wildcard/static/src/js/search.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+odoo.define('web_advanced_search_wildcard', function (require) {
```
Comment (elicoidal) [inline web_advanced_search_wildcard/static/src/js/search.js:1]: You could update copyright here

File: web_advanced_search_wildcard/views/template.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
```
Comment (elicoidal) [inline web_advanced_search_wildcard/views/template.xml:2]: and here

**Review by elicoidal — APPROVED** [review]
LGTM

File: web_advanced_search_wildcard/README.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+Credits
+=======
+
+Images
+------
+
+* Odoo Community Association: `Icon <https://odoo-community.org/logo.png>`_.
+
+Contributors
```
Comment (Tardo) [inline web_advanced_search_wildcard/README.rst:39]: Change this to the new structure for readme's

**Review by Tardo — CHANGES_REQUESTED** [review]

**Comment by Tardo** [comment]
Despite the necessary changes, it seems to work correctly.

File: web_advanced_search_wildcard/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Wildcard in advanced search",
+    "summary": "Webmodule to add wildcard operators in advanced search field",
+    "version": "11.0.1.0.0",
+    "category": "web",
+    "license": "AGPL-3",
+    "author": "initOS GmbH & Co. KG, Therp BV, "
+              "Odoo Community Association (OCA)",
+    "application": False,
```
Comment (simahawk) [inline web_advanced_search_wildcard/__manifest__.py:13]: drop this flags, as they are defaults. 
Add `'website': 'https://github.com/OCA/web',`

**Review by simahawk — APPROVED** [review]
only small changes needed

File: web_advanced_search_wildcard/static/src/js/search.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    "use strict";
+
+    var core = require('web.core');
+    var _lt = core._lt;
+    var Field = core.search_filters_registry.get("char");
+
+    Field.prototype.operators.push(
```
Comment (tarteo) [inline web_advanced_search_wildcard/static/src/js/search.js:8]: Can you use the class system here, something like:

```
Char.include({
      init: function() {
            this._super.apply(this, arguments);
            this.operators.push({value: '=ilike', text: _lt("matches")});
      }
})

```

**Review by tarteo — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
@JBF91 can you make the requested comments?

**Comment by JBF91** [comment]
@pedrobaeza Yes, I'm going to do it these days

**Review by alexis-via — APPROVED** [review]
code review, no test

**Comment by alexis-via** [comment]
If you like it, you can include the improvement is proposed on the v10 branch https://github.com/OCA/web/pull/1142/files (adds "starts with" and "ends with" advanced search filters).

**Comment by simahawk** [comment]
@JBF91 any update?

**Comment by nachoserra-domatix** [comment]
Hi, there seems to be an incompatibility when the module web_advanced_search is installed and you try to filter a many2one field. For example, in journal items, searching accounts starting with '%6' it returns journal items where the account id is 6.

**Pull request closed without merging**